client/lib/mixins/i18n/index.js

```diff
@@ -14,7 +14,7 @@ var debug = require( 'debug' )( 'calypso:i18n' ),
 var config = require( 'config' ),
 	numberFormatPHPJS = require( './number-format' ),
 	emitter = require( 'lib/mixins/emitter' ),
-	interpolateComponents = require( 'lib/interpolate-components' );
+	interpolateComponents = require( 'lib/interpolate-components/src/' );
```

Comment:
Requiring from a sub-directory in a published module will not find the package.json, the main reason to require it to install dependencies.  I did not explain correctly how this PR works.  Please see the main description which I have updated.